website/components/remote-plugin-docs/utils/merge-remote-plugins.js

```diff
@@ -0,0 +1,166 @@
+const path = require('path')
+const fetchGithubFile = require('./fetch-github-file')
+
+const COMPONENT_TYPES = [
+  'builders',
+  'datasources',
+  'post-processors',
+  'provisioners',
+]
+
+async function gatherRemotePlugins(pluginsData, navData, isDev = true) {
+  const allPluginData = await Promise.all(
+    pluginsData.map(async (pluginEntry) => {
+      const componentEntries = await Promise.all(
+        COMPONENT_TYPES.map(async (type) => {
+          const routes = await gatherPluginBranch(pluginEntry, type)
+          if (!routes) return false
+          const isSingleLeaf =
+            routes.length === 1 && typeof routes[0].path !== 'undefined'
+          const navData = isSingleLeaf
+            ? { ...routes[0], path: path.join(type, pluginEntry.path) }
+            : { title: pluginEntry.title, routes }
+          return { type, navData }
+        })
+      )
+      const validComponents = componentEntries.filter(Boolean)
+      if (validComponents.length === 0) {
+        const errMsg = `Could not fetch any component documentation for remote plugin from ${pluginEntry.repo}. This may be a GitHub credential issue at build time, or it may be an issue with missing docs in the source repository. Please ensure you have a valid GITHUB_API_TOKEN set in .env.local at the root of the project.`
+        if (isDev) {
+          console.warn(errMsg)
+        } else {
+          throw new Error(errMsg)
```

Comment:
In production, we don't want to build the site with missing content. In dev, we don't want to force authors to have a `GITHUB_API_TOKEN` just to preview any other part of the site, so we just warn that there is missing content, both here in standard output and in a little preview bar in the actual site.